frontend/pages/packs/EditPackPage/EditPackPage.tsx

```diff
@@ -0,0 +1,371 @@
+  } = useQuery<IStoredTeamsResponse, Error, ITeam[]>(
+    ["pack labels"],
+    () => teamsAPI.loadAll(),
+    {
+      select: (data: IStoredTeamsResponse) => data.teams,
+    }
+  );
+
+  const packTeams = storedPack
+    ? filter(teams, (team) => {
+        return includes(storedPack.team_ids, team.id);
+      })
+    : [];
+
+  const packTargets = [...packHosts, ...packLabels, ...packTeams];
+
+  const onCancelEditPack = () => {
+    return dispatch(push(PATHS.MANAGE_PACKS));
+  };
+
+  const onFetchTargets = (
+    query: IQuery,
+    targetsResponse: ITargetsAPIResponse
+  ) => {
+    const { targets_count } = targetsResponse;
+
+    setTargetsCount(targets_count);
+
+    return false;
+  };
+
+  const togglePackQueryEditorModal = useCallback(() => {
+    setSelectedPackQuery(undefined); // create modal renders
+    setShowPackQueryEditorModal(!showPackQueryEditorModal);
+    console.log("togglePackQueryEditorModal clicked!");
+  }, [showPackQueryEditorModal, setShowPackQueryEditorModal]);
+
+  const toggleEditPackQueryModal = useCallback(() => {
+    setShowEditPackQueryModal(!showEditPackQueryModal);
+  }, [showEditPackQueryModal, setShowEditPackQueryModal]);
+
+  const toggleRemovePackQueryModal = useCallback(() => {
+    setShowRemovePackQueryModal(!showRemovePackQueryModal);
+  }, [showRemovePackQueryModal, setShowRemovePackQueryModal]);
+
+  const onEditPackQueryClick = (selectedQuery: any): void => {
+    togglePackQueryEditorModal();
+    setSelectedPackQuery(selectedQuery); // edit modal renders
+  };
+
+  const onRemovePackQueriesClick = (selectedTableQueryIds: any): void => {
+    toggleRemovePackQueryModal();
+    setSelectedPackQueryIds(selectedTableQueryIds);
+  };
+
+  const handlePackFormSubmit = (formData: any) => {
+    const updatedPack = deepDifference(formData, storedPack);
+    packsAPI
+      .update(packId, updatedPack)
+      .then(() => {
+        dispatch(renderFlash("success", `Successfully updated this pack.`));
+      })
+      .catch(() => {
+        dispatch(
+          renderFlash("error", `Could not update pack. Please try again.`)
+        );
+      });
+  };
+
+  const onPackQueryEditorSubmit = useCallback(
+    (formData: IPackQueryFormData, editQuery: IScheduledQuery | undefined) => {
+      const request = editQuery
+        ? scheduledqueriesAPI.update(editQuery, formData)
+        : scheduledqueriesAPI.create(formData);
+      request
+        .then(() => {
+          dispatch(renderFlash("success", `Successfully updated this pack.`));
+        })
+        .catch(() => {
+          dispatch(
+            renderFlash(
+              "error",
+              "Could not update this pack. Please try again."
+            )
+          );
+        })
+        .finally(() => {
+          togglePackQueryEditorModal();
+          getPackQueries();
+        });
+      return false;
+    },
+    [dispatch, getPackQueries, togglePackQueryEditorModal]
+  );
+
+  const onRemovePackQuerySubmit = useCallback(() => {
```

Comment:
I don't think useCallback is necessary. The function is run in a controlled manner.